include/aspect/particle/particle.h

```diff
@@ -21,119 +21,141 @@
-        Point<dim>      location;
-
-        /**
-         * Current particle velocity
-         */
-        Point<dim>      velocity;
-
-        /**
-         * Globally unique ID of particle
-         */
-        double          id;
-
-        /**
-         * Whether this particle is in the local subdomain or not
-         */
-        bool            is_local;
-
-        /**
-         * Whether to check the velocity of this particle This is used for
-         * integration schemes which require multiple integration steps for
-         * some particles, but not for others
-         */
-        bool            check_vel;
-
       public:
         /**
-         * Empty constructor for BaseParticle, creates a particle at the
-         * origin with zero velocity.
+         * Empty constructor for Particle, creates a particle at the
+         * origin.
          */
-        BaseParticle ();
+        Particle ();
 
         /**
-         * Constructor for BaseParticle, creates a particle with the specified
-         * ID at the specified location with zero velocity. Note that Aspect
-         * does not check for duplicate particle IDs so the user must be sure
-         * the IDs are unique over all processes.
+         * Constructor for Particle, creates a particle with the specified
+         * ID at the specified location. Note that Aspect
+         * does not check for duplicate particle IDs so the generator must
+         * make sure the IDs are unique over all processes.
          *
          * @param[in] new_loc Initial location of particle.
          * @param[in] new_id Globally unique ID number of particle.
          */
-        BaseParticle (const Point<dim> &new_loc,
-                      const double &new_id);
+        Particle (const Point<dim> &new_loc,
+                  const particle_index &new_id);
+
+        /**
+         * Constructor for Particle, creates a particle from a data vector.
+         * This constructor is usually called after sending a particle to a
+         * different process.
+         *
+         * @param[in,out] begin_data First data component.
+         * @param[in] data_len Number of components of the begin_data vector
+         * that will be read in by this particle.
+         */
+        Particle (const void *&begin_data,
+                  const unsigned int data_len);
 
         /**
-         * Destructor for BaseParticle
+         * Destructor for Particle
          */
         virtual
-        ~BaseParticle ();
+        ~Particle ();
 
         /**
          * Get the number of doubles required to represent this particle for
-         * communication.
+         * communication. This includes the base properties like position
+         * and id as well as the additional user requested properties.
          *
          * @return Number of doubles required to represent this particle
          */
-        static unsigned int
-        data_len ();
+        unsigned int
+        data_len () const;
```

Comment:
I don't particularly care for this name. How about `n_particle_properties_for_serialization()`? Or more briefly `n_props_for_serialization()`?